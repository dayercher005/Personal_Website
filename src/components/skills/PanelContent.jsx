import HTML from '/public/HTML.svg'
import CSS from '/public/CSS.svg'
import Javascript from '/public/Javascript.svg'
import Python from '/public/Python.svg'
import Tailwind from '/public/Tailwind.svg'
import React from '/public/React.svg'
import NodeJS from '/public/NodeJS.svg'
import Webpack from '/public/Webpack.svg'
import Vite from '/public/Vite.svg'
import Git from '/public/Git.svg'

export default function PanelContent({currentIndex}) {

    const ProgrammingLanguages = [
        {name: 'HTML', image: HTML},
        {name: 'CSS', image: CSS},
        {name: 'Javascript', image: Javascript},
        {name: 'Python', image: Python}
    ]

    const Frontend = [
        {name: 'Tailwind', image: Tailwind},
        {name: 'React', image: React}
    ]

    const Backend = [
        {name: 'NodeJS', image: NodeJS}
    ]

    const Tools = [
        {name: 'Webpack', image: Webpack},
        {name: 'Vite', image: Vite},
        {name: 'Git', image: Git}
    ]

    const ProgrammingLanguageIndividualContent = ProgrammingLanguages.map(language => 
        <div className = "flex rounded-xl bg-gray-200 p-3 middle-align">
            <img className = "size-10" src={language.image}/>
            <p className = "text-xl font-medium">{language.name}</p>
        </div>
    )

    const FrontendContent = Frontend.map(framework => 
        <div className="flex">
            <img src={framework.image}/>
            <p>{framework.name}</p>
        </div>
    )

    const BackendContent = Backend.map(framework => 
        <div className="flex">
            <img src={framework.image}></img>
            <p>{framework.name}</p>
        </div>
    )

    const ToolsContent = Tools.map(tools => 
        <div className="flex">
            <img src={tools.image}></img>
            <p>{tools.name}</p>
        </div>
    )

    const ProgrammingLanguageContent = (
        <div className="grid grid-cols-3 w-1/3 mx-auto gap-15">
            {ProgrammingLanguageIndividualContent}
        </div>
    )

    if (currentIndex === 0){
        return (
            ProgrammingLanguageContent
        )
    } else if (currentIndex === 1){
        return (
            FrontendContent
        )
    } else if (currentIndex === 2){
        return (
            BackendContent
        )
    } else if (currentIndex === 3){
        return (
            ToolsContent
        )
    }
    
}