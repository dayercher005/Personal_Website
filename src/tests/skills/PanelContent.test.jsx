import { describe, it, expect } from 'vitest'
import { render } from "@testing-library/react";
import PanelContent from '/src/components/skills/PanelContent.jsx'

describe("PanelContent component under Skills component", () => {
    it("should render respective Panel Content based on user Click", () => {
        const { container } = render(<PanelContent/>)
        expect(container).toMatchSnapshot()
    })
})