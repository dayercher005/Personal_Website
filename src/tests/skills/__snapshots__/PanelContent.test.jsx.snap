// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PanelContent component under Skills component > should render respective Panel Content based on user Click 1`] = `<div />`;
